vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/MergedRoomCreationItem.kt

```diff
@@ -107,13 +96,121 @@ abstract class MergedRoomCreationItem : BasedMergedItem<MergedRoomCreationItem.H
         holder.readReceiptsView.isVisible = false
     }
 
+    private fun bindEncryptionTile(holder: Holder, data: Data?) {
+        if (attributes.hasEncryptionEvent) {
+            holder.encryptionTile.isVisible = true
+            holder.encryptionTile.updateLayoutParams<ConstraintLayout.LayoutParams> {
+                this.marginEnd = leftGuideline
+            }
+            if (attributes.isEncryptionAlgorithmSecure) {
+                holder.e2eTitleTextView.text = holder.expandView.resources.getString(R.string.encryption_enabled)
+                holder.e2eTitleDescriptionView.text = if (data?.isDirectRoom == true) {
+                    holder.expandView.resources.getString(R.string.direct_room_encryption_enabled_tile_description)
+                } else {
+                    holder.expandView.resources.getString(R.string.encryption_enabled_tile_description)
+                }
+                holder.e2eTitleDescriptionView.textAlignment = View.TEXT_ALIGNMENT_CENTER
+                holder.e2eTitleTextView.setCompoundDrawablesWithIntrinsicBounds(
+                        ContextCompat.getDrawable(holder.view.context, R.drawable.ic_shield_black),
+                        null, null, null
+                )
+            } else {
+                holder.e2eTitleTextView.text = holder.expandView.resources.getString(R.string.encryption_not_enabled)
+                holder.e2eTitleDescriptionView.text = holder.expandView.resources.getString(R.string.encryption_unknown_algorithm_tile_description)
+                holder.e2eTitleTextView.setCompoundDrawablesWithIntrinsicBounds(
+                        ContextCompat.getDrawable(holder.view.context, R.drawable.ic_shield_warning),
+                        null, null, null
+                )
+            }
+        } else {
+            holder.encryptionTile.isVisible = false
+        }
+    }
+
+    private fun bindCreationSummaryTile(holder: Holder) {
```

Comment:
I find there is too much logic on the item which should be done in the controller